tests/testmultistream.nim

```diff
@@ -107,33 +118,39 @@ type
     step*: int
     na*: NaHandler
 
-method readExactly*(s: TestNaStream,
-                    pbytes: pointer,
-                    nbytes: int):
-                    Future[void] {.async, gcsafe.} =
+method readOnce*(s: TestNaStream,
+                 pbytes: pointer,
+                 nbytes: int):
```

Comment:
technically, this test is not correct any more - ie it would be valid for a caller to readonce with `nbytes=1` lots of times here and it should still pass the test

Reply:
Need to revisit it apparently.

Reply:
I'm not sure if you mean for that test specifically or for all the tests in this file? If you're talking about all the tests, then this are simply data mocks that yield a deterministic value on each step. They are multistream specific.

We should test the streaming functionality on it's own, like we do for bufferstream for example.